File: benchmarks/object_allocation/object_allocation.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
  long long value;
} Box;

Box *box_new(long long v) {
  Box *b = (Box *) malloc(sizeof(Box));
  b->value = v;
  return b;
}

int main(void) {
  long long total = 0;
  long long i = 0;
  while (i < 1000000LL) {
    Box *b = box_new(i);
    total += b->value;
    free(b);
    i += 1;
  }
  printf("%lld\n", total);
  return 0;
}

File: benchmarks/object_allocation/object_allocation.rs
struct Box2 {
  value: i64,
}

fn main() {
  let mut total: i64 = 0;
  let mut i: i64 = 0;
  while i < 1_000_000 {
    let b = Box::new(Box2 { value: i });
    total += b.value;
    i += 1;
  }
  println!("{total}");
}
